### PR TITLE
migrate snap to core24 + update minimum python version to 3.10

### PR DESCRIPTION
**Pull request by rthill91**

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -50,11 +51,7 @@ apps:
 
 parts:
   awspub:
-    plugin: python
+    plugin: poetry
```
Comment (toabctl) [inline snap/snapcraft.yaml:54]: unrelated to the commit message.

**Review by toabctl — CHANGES_REQUESTED** [review]
please split those commits
* use poetry plugin
* snap: use core24
* bump minimal python version
*

**Review by rpocase — DISMISSED** [review]
lgtm pending tom's requested changes

**Review by toabctl — CHANGES_REQUESTED** [review]
we try to follow conventional commits (see `git log`) so please add commit types and also a brief reasoning for each commit.

**Review by JessicaJang — APPROVED** [review]

**Review by toabctl — APPROVED** [review]

**Pull request merged**